lib/addons/backstage/index.ts

```diff
@@ -53,51 +52,65 @@ export class BackstageAddOn extends HelmAddOn {
     const chart = this.addHelmChart(clusterInfo, values);
 
+    new CfnOutput(clusterInfo.cluster.stack, 'Backstage base URL', {
+      value: HTTPS+this.options.subdomain,
+      description: "Backstage base URL",
+      exportName: "BackstageBaseUrl",
+    });
+
     return Promise.resolve(chart);
   }
-}
 
-/**
- * populateValues populates the appropriate values used to customize the Helm chart
- * @param helmOptions User provided values to customize the chart
- */
-function populateValues(clusterInfo: ClusterInfo, helmOptions: BackstageAddOnProps): Values {
-  const values = helmOptions.values ?? {};
-  
-  const annotations = {
-    "alb.ingress.kubernetes.io/scheme": "internet-facing",
-    "alb.ingress.kubernetes.io/target-type": "ip",
-    "alb.ingress.kubernetes.io/certificate-arn": clusterInfo.getResource<ICertificate>(helmOptions.certificateResourceName)?.certificateArn
-  };
+  /**
+  * populateValues populates the appropriate values used to customize the Helm chart
+  * @param helmOptions User provided values to customize the chart
+  */
+  populateValues(clusterInfo: ClusterInfo, helmOptions: BackstageAddOnProps): Values {
+    const values = helmOptions.values ?? {};
+    
+    const annotations = {
+      "alb.ingress.kubernetes.io/scheme": "internet-facing",
+      "alb.ingress.kubernetes.io/target-type": "ip",
+      "alb.ingress.kubernetes.io/certificate-arn": clusterInfo.getResource<ICertificate>(helmOptions.certificateResourceName)?.certificateArn
+    };
   
-  const database = {
-    "client": "pg",
-    "connection": {
-      "host": helmOptions.postgresHost,
-      "port": helmOptions.postgresPort,
-      "user": helmOptions.postgresUser,
-      "password": helmOptions.postgresPassword,
+    const databaseInstance: rds.IDatabaseInstance  = clusterInfo.getRequiredResource(helmOptions.databaseResourceName);
+    if (databaseInstance === undefined) {
+        throw new Error("Database instance not found");
     }
-  };
   
-  setPath(values, "ingress.enabled", true);
-  setPath(values, "ingress.className", "alb");
-  setPath(values, "ingress.host", helmOptions.subdomain);
-  setPath(values, "ingress.annotations", annotations);
-
-  setPath(values, "backstage.image.registry", helmOptions.imageRegistry);
-  setPath(values, "backstage.image.repository", helmOptions.imageRepository);
-  setPath(values, "backstage.image.tag", helmOptions.imageTag);
-
-  setPath(values, "backstage.appConfig.app.baseUrl", helmOptions.baseUrl);
-  setPath(values, "backstage.appConfig.backend.baseUrl", helmOptions.baseUrl);
-  setPath(values, "backstage.appConfig.backend.database", database);
+    const databaseChartValues = {
+      "client": "pg",
+      "connection": {
+        "host": databaseInstance.dbInstanceEndpointAddress,
+        "port": databaseInstance.dbInstanceEndpointPort,
+        "user": "${POSTGRES_USER}",
+        "password": "${POSTGRES_PASSWORD}"
+      }
+    };
```

Comment:
As far as I could understand from the backstage chart, these values are supposed to be keys to K8s secrets and not the raw values themselves. Please let me know if my understanding is wrong.

Reply:
Correct, see here: https://backstage.io/docs/tutorials/switching-sqlite-postgres/#add-postgresql-configuration
POSTGRES_USER and POSTGRES_PASSWORD are here 2 environment variables which will be populated from a Secret. Please Dependencies section in backstage.md here and Secret creation in lib/backstage-construct/backstage-secret-addon.ts from this other PR: https://github.com/aws-samples/cdk-eks-blueprints-patterns/pull/81/files